recipes/asio-grpc/all/conanfile.py

```diff
@@ -0,0 +1,82 @@
+from conans import ConanFile, CMake, tools
+import os
+
+required_conan_version = ">=1.43.0"
+
+
+class AsioGrpcConan(ConanFile):
+    name = "asio-grpc"
+    description = ("Asynchronous gRPC with Asio/unified executors")
+    homepage = "https://github.com/Tradias/asio-grpc"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "Apache-2.0"
+    topics = ("cpp", "asynchronous", "grpc", "asio", "asynchronous-programming", "cpp17", "coroutine", "cpp20", "executors")
+    settings = "compiler"
```

Comment:
```suggestion
    settings = "compiler", "build_type", "os", "arch"
```
Keep full settings, change the package id.

Reply:
A header-only library does not change based on those settings. Is that already taken care of by `self.info.header_only()` or do I need to e.g. `del self.settings.arch` inside `package_id()`?

Reply:
Yes it does! (the docs around this method are really weak)

The adding all setting is just a convention for CCI as I recall, there was some technical discussion but it's been too long for me to recall